app/assets/javascripts/components/students/assign_button.jsx

```diff
@@ -123,7 +123,12 @@ const AssignButton = React.createClass({
       return;
     }
 
+    const closePopup = this.props.open;
```

Comment:
A comment here would be useful, explaining that it should only close the popup if it's the 'add available' version of this button.